yt_dlp/extractor/manyvids.py

```diff
@@ -1,114 +1,98 @@
+                'thumbnail': 'https://ods.manyvids.com/1001061960/3aa5397f2a723ec4597e344df66ab845/screenshots/thumbs/custom_1_180_5be09c1dcce03.jpg',
+                'uploader': 'Sarah Calanthe',
+                'view_count': int,
+                'like_count': int,
+            },
+        },
+    ]
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
 
-        real_url = f'https://www.manyvids.com/video/{video_id}/gtm.js'
-        try:
-            webpage = self._download_webpage(real_url, video_id)
-        except Exception:
-            # probably useless fallback
-            webpage = self._download_webpage(url, video_id)
-
-        info = self._search_regex(
-            r'''(<div\b[^>]*\bid\s*=\s*(['"])pageMetaDetails\2[^>]*>)''',
-            webpage, 'meta details', default='')
-        info = extract_attributes(info)
+        info = traverse_obj(
+            self._download_json(f'https://www.manyvids.com/bff/store/video/{video_id}', video_id),
+            ('data', {dict})) or {}
 
-        player = self._search_regex(
-            r'''(<div\b[^>]*\bid\s*=\s*(['"])rmpPlayerStream\2[^>]*>)''',
-            webpage, 'player details', default='')
-        player = extract_attributes(player)
+        video_urls = traverse_obj(
+            self._download_json(f'https://www.manyvids.com/bff/store/video/{video_id}/private', video_id),
+            ('data', {dict})) or {}
 
         video_urls_and_ids = (
-            (info.get('data-meta-video'), 'video'),
-            (player.get('data-video-transcoded'), 'transcoded'),
-            (player.get('data-video-filepath'), 'filepath'),
-            (self._og_search_video_url(webpage, secure=False, default=None), 'og_video'),
+            (traverse_obj(video_urls, ('teaser', 'filepath')), 'preview'),
+            (video_urls.get('transcodedFilepath'), 'transcoded'),
+            (video_urls.get('filepath'), 'filepath'),
         )
 
-        def txt_or_none(s, default=None):
-            return (s.strip() or default) if isinstance(s, str) else default
-
-        uploader = txt_or_none(info.get('data-meta-author'))
-
-        def mung_title(s):
-            if uploader:
-                s = re.sub(rf'^\s*{re.escape(uploader)}\s+[|-]', '', s)
-            return txt_or_none(s)
-
-        title = (
-            mung_title(info.get('data-meta-title'))
-            or self._html_search_regex(
-                (r'<span[^>]+class=["\']item-title[^>]+>([^<]+)',
-                 r'<h2[^>]+class=["\']h2 m-0["\'][^>]*>([^<]+)'),
-                webpage, 'title', default=None)
-            or self._html_search_meta(
-                'twitter:title', webpage, 'title', fatal=True))
-
-        title = re.sub(r'\s*[|-]\s+ManyVids\s*$', '', title) or title
-
-        if any(p in webpage for p in ('preview_videos', '_preview.mp4')):
+        def convert_string_suffix_num(num_str):
+            multipliers = {'k': 1000, 'm': 1000000}
```

Comment:
The API does not return int values for the views and like count, but stringified numbers such as "123k".